src/main/java/li/cil/sedna/riscv/R5CPUTemplate.java

```diff
@@ -778,6 +778,21 @@ private boolean writeCSR(final int csr, final long value) throws R5IllegalInstru
                         }
                     }
 
+                    // From RISC-V Privileged Architectures, section 4.2.1
+                    // "Changing satp.MODE from Bare to other modes and vice versa also takes effect immediately,
+                    // without the need to execute an SFENCE.VMA instruction."
+                    if (xlen == R5.XLEN_32) {
+                        if ((satp & R5.SATP_MODE_MASK32) == R5.SATP_MODE_NONE ||
```

Comment:
Does this also apply to PPN changes? If not, would it make more sense to just do this for to/from bare mode changes? I.e. something like
```java
//                                                      vv
if (((satp & R5.SATP_MODE_MASK32) == R5.SATP_MODE_NONE) !=
    ((validatedValue & R5.SATP_MODE_MASK32) == R5.SATP_MODE_NONE))
```
instead?

Reply:
I agree, that makes more sense. Even though the other bits should always be 0 in Bare mode.

Reply:
Ah, good point! Thanks.